### PR TITLE
Fixed overwriting downloaded files in headless mode

### PR DESCRIPTION
**Pull request by StevenWuzz**
### :scroll: Description

- Describe the general shape of this PR (new feature? refactor? bug fix? one-line change?)
Bug fix

- Describe what changes are being made
Modified scraper.py such that it renames the downloaded transactionExportGrid.xls file into transactionExportGrid.xls(1) or transactionExportGrid.xls(2) and so on. 

- Describe why these changes are being made
To avoid overwriting downloaded files in the headless mode

- List the use cases and edge cases relevant to this PR

- Describe how errors will be handled. How will we know if this code breaks in production

- Describe any external libraries/dependencies added or removed in this PR
None

- Describe any security risks are there regarding this change

- Describe how you tested these changes

- Link to relevant external documentation

--------------------------
### :clipboard: Mandatory Checklist

- [x] Linked to the Github Issues being addressed using the right sidebar :arrow_right:
- [x] Have you discussed these changes with the project leader(s)?
- [x] Do all variable and function names communicate what they do?
- [] Were all the changes commented and / or documented?
- [x] Is the PR the right size? (If the PR is too large to review, it might be good to break it up into multiple PRs.)
- [] Does all work in progress, temporary, or debugger code have a TODO comment with links to Github issues?
- [] *If you changed the user interface, did you add before and after screenshots to below?*

--------------------------
#### After

--------------------------
### :blue_book: Glossary
- PR = Pull Request

### CODE REVIEW

**Review by rwalek668 — CHANGES_REQUESTED** [review]

File: data_pipeline/scraper/scraper.py [hunk 1]
```diff
@@ -117,8 +118,16 @@ def downloadExcel(self, driver):
                 else:
                     downloadLinkElement.click()
                     count += 1
+                    sleep(1)
```
[the remark this answers is not in the document]
Comment (StevenWuzz) [reply data_pipeline/scraper/scraper.py:121]: Yep, I have done that. Based on my observation, I don't think it has caused an extreme slowdown such as 10x slower. It's definitely slightly slower, but it's not really critical (at least in my opinion). However, I will try to find an alternative solution for that. Thanks!

File: data_pipeline/scraper/scraper.py [hunk 2]
```diff
@@ -230,11 +239,12 @@ def __init__(self):
 
         # Uncomment block BELOW for headless data-retrieval
         # --> Currently not working 100%, only downloads first link on form table
-        isHeadless = os.environ.get('HEADLESS', False)
+        #isHeadless = os.environ.get('HEADLESS', False)
```
[the remark this answers is not in the document]
Comment (StevenWuzz) [reply data_pipeline/scraper/scraper.py:242]: Got it. I'll do that. Thanks!

File: data_pipeline/scraper/scraper.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (StevenWuzz) [reply data_pipeline/scraper/scraper.py:131]: Update: I've modified the solution

**Comment by StevenWuzz** [comment]
@rwalek668 could you take a look at it again? Thank you!

File: data_pipeline/scraper/scraper.py [hunk 3]
```diff
@@ -117,8 +118,27 @@ def downloadExcel(self, driver):
...
+                    
+                    while(1):
+                        if os.path.exists('./data/transactionExportGrid.xls'):
+                            countFile += 1
+                            renamedFile = './data/transactionExportGrid' + '(' + str(countFile) + ').xls'
+                            os.rename('./data/transactionExportGrid.xls', renamedFile)
+                            break
+                        sleep(0.1)
+                    
+                    """
```
Comment (rwalek668) [inline data_pipeline/scraper/scraper.py:130]: You can delete the commented out code, then looks good to submit.

**Review by rwalek668 — APPROVED** [review]

File: data_pipeline/scraper/scraper.py (hunk 3 above)
Comment (StevenWuzz) [reply data_pipeline/scraper/scraper.py]: Got it! Sorry, I somehow didn't realize that I left it commented there

**Pull request merged**